thermochem/janaf.py

```diff
@@ -97,27 +108,48 @@ def __init__(self, rawdata_text):
         data['Delta_fH'] *= 1000
         data['Delta_fG'] *= 1000
 
+        # Handle NaNs for the phase transition points.  This only affects Delta_fG, Delta_fH, and log(Kf)
+        GoodIndices = np.where(np.isfinite(data['Delta_fH']))
+        print(GoodIndices)
+
         # Now make interpolatable functions for each of these.
         self.cp = interp1d(self.rawdata['T'], self.rawdata['Cp'])
         self.S = interp1d(self.rawdata['T'], self.rawdata['S'])
         self.gef = interp1d(self.rawdata['T'], self.rawdata['[G-H(Tr)]/T'])
         self.hef = interp1d(self.rawdata['T'], self.rawdata['H-H(Tr)'])
-        self.DeltaH = interp1d(self.rawdata['T'], self.rawdata['Delta_fH'])
-        self.DeltaG = interp1d(self.rawdata['T'], self.rawdata['Delta_fG'])
-        self.logKf = interp1d(self.rawdata['T'], self.rawdata['log(Kf)'])
+        self.DeltaH = interp1d(self.rawdata['T'].iloc[GoodIndices], self.rawdata['Delta_fH'].iloc[GoodIndices])
+        self.DeltaG = interp1d(self.rawdata['T'].iloc[GoodIndices], self.rawdata['Delta_fG'].iloc[GoodIndices])
+        self.logKf = interp1d(self.rawdata['T'].iloc[GoodIndices], self.rawdata['log(Kf)'].iloc[GoodIndices])
 
         # TODO Deal well with crystal<->liquid transitions which have a below
         # and above value for Cp, S, etc.
 
+    def __str__(self):
+        rep = super().__str__()
+        rep += "\n  "
+        rep += self.description
+        rep += "\n    Cp(%0.2f) = %0.3f J/mol/K"%(Tr, self.cp(Tr))
+        rep += "\n    S(%0.2f) = %0.3f J/mol/K"%(Tr, self.S(Tr))
+        rep += "\n    [G-H(%0.2f)]/%0.2f = %0.3f J/mol/K"%(Tr, Tr, self.gef(Tr))
+        rep += "\n    H-H(%0.2f) = %0.3f J/mol/K"%(Tr, self.hef(Tr))
+        rep += "\n    Delta_fH(%0.2f) = %0.0f J/mol"%(Tr, self.DeltaH(Tr))
+        rep += "\n    Delta_fG(%0.2f) = %0.0f J/mol"%(Tr, self.DeltaG(Tr))
+        rep += "\n    log(Kf((%0.2f)) = %0.3f"%(Tr, self.logKf(Tr))
+        return rep
```

Comment:
This is so helpful

Reply:
Thanks!  I mostly wanted it for myself.  ;-)